clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -587,6 +587,19 @@ void resubscribeForAllPartitions() throws InterruptedException {
     }
   }
 
+  /**
+   * Removes partitions that are not subscribed / not assigned.
+   */
+
+  public synchronized void removeUnsubscribedPartitions() {
+    throwIfNotRunning();
+    for (PartitionConsumptionState partitionConsumptionState: partitionConsumptionStateMap.values()) {
+      if (!partitionConsumptionState.isSubscribed()) {
+        partitionConsumptionStateMap.remove(partitionConsumptionState.getPartition(), partitionConsumptionState);
+      }
+    }
```

Comment:
I believe in the meeting we discussed that this isn't the correct way to check for partitions to be removed. Please correct me if I'm wrong.